processing/src/main/java/io/druid/segment/filter/InFilter.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.segment.filter;
+
+import com.google.common.base.Predicate;
+import com.google.common.base.Strings;
+import com.google.common.collect.Sets;
+import com.metamx.collections.bitmap.ImmutableBitmap;
+import io.druid.query.dimension.DefaultDimensionSpec;
+import io.druid.query.filter.BitmapIndexSelector;
+import io.druid.query.filter.Filter;
+import io.druid.query.filter.ValueMatcher;
+import io.druid.query.filter.ValueMatcherFactory;
+import io.druid.segment.ColumnSelectorFactory;
+import io.druid.segment.DimensionSelector;
+import io.druid.segment.data.IndexedInts;
+
+import java.util.Set;
+
+/**
+ */
+public class InFilter implements Filter
+{
+  private final String dimension;
+  private final Set<String> values;
+
+  public InFilter(String dimension, Set<String> values)
+  {
+    this.dimension = dimension;
+    this.values = values;
+  }
+
+  @Override
+  public ImmutableBitmap getBitmapIndex(BitmapIndexSelector selector)
+  {
+    ImmutableBitmap bitmap = selector.getBitmapFactory().makeEmptyImmutableBitmap();
+    for (String value : values) {
+      bitmap = bitmap.union(selector.getBitmapIndex(dimension, value));
+    }
+    return bitmap;
+  }
+
+  @Override
+  public ValueMatcher makeMatcher(ValueMatcherFactory factory)
+  {
+    return factory.makeValueMatcher(
+        dimension, new Predicate<String>()
+        {
+          @Override
+          public boolean apply(String input)
+          {
+            return values.contains(Strings.nullToEmpty(input));
```

Comment:
`values` may contain null, this should match `""` but with the current code it will not. Something should be translating null to empty within `values`.